src/PermissionManager.php

```diff
@@ -32,32 +28,37 @@ class PermissionManager implements PermissionManagerInterface {
   protected $entityTypeBundleInfo;
 
   /**
+   * The event dispatcher.
+   *
+   * @var \Symfony\Component\EventDispatcher\EventDispatcherInterface
+   */
+  protected $eventDispatcher;
+
+  /**
    * Constructs a PermissionManager object.
    *
-   * @param \Drupal\og\GroupManager $group_manager
-   *   The OG group manager.
    * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
    *   The entity type manager.
    * @param \Drupal\Core\Entity\EntityTypeBundleInfoInterface $entity_type_bundle_info
    *   The service providing information about bundles.
+   * @param \Symfony\Component\EventDispatcher\EventDispatcherInterface $event_dispatcher
+   *   The event dispatcher.
    */
-  public function __construct(GroupManager $group_manager, EntityTypeManagerInterface $entity_type_manager, EntityTypeBundleInfoInterface $entity_type_bundle_info) {
-    $this->groupManager = $group_manager;
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, EntityTypeBundleInfoInterface $entity_type_bundle_info, EventDispatcherInterface $event_dispatcher) {
     $this->entityTypeManager = $entity_type_manager;
     $this->entityTypeBundleInfo = $entity_type_bundle_info;
+    $this->eventDispatcher = $event_dispatcher;
   }
 
   /**
    * {@inheritdoc}
-   *
-   * @todo Provide an alter hook.
    */
-  public function getPermissionList($entity_type_id, $bundle_id) {
+  public function getEntityOperationPermissions(array $group_content_bundle_ids) {
     $permissions = [];
 
-    foreach ($this->groupManager->getGroupContentBundleIdsByGroupBundle($entity_type_id, $bundle_id) as $group_content_entity_type_id => $group_content_bundle_ids) {
```

Comment:
This was the circular dependency: here the `GroupManager` was called again. This is now solved by passing in the group content bundle IDs.